public/js/scripts.js

```diff
@@ -1,56 +1,115 @@
-                    results.innerHTML += "<p>Publisher: " + data.items[i].volumeInfo.publisher + "</p>"
-                } else {
-                    results.innerHTML +=  "<p>Publisher: Sorry, this content is not available. Please check out the preview link below.</p>"
+// AJAX request function, param stores search query results  - AJAX or Asynchronous JavaScript And XML is the use of the XMLHttpRequest object to communicate with servers without reloading the page. The AJAX request itself has 4 server responses /status properties `readyState`, `onreadystatechange`, `status`, `statusText`
+const asyncReq = (searchQuery) => {
+  // create promise instance using `Promise()` constructor. - promise is returned on `asyncReq` - Promise(((executor functions)
+    return new Promise (((resolve,reject) => {
+      // XMLHttpRequest object is created
+            const xhr = new XMLHttpRequest();
+            // callback runs each time there is a change of the ajax request - .onreadystatechange property (contains an event handler `readystatechange`) defines a function to be called when the readyState property changes
+        xhr.onreadystatechange = (() => {
+          // readyState holds status of XMLHttpRequest object, 4 = request finished and response is ready
+                if (xhr.readyState == 4) {
+                  // 200 = success status code
+                    if (xhr.status == 200) {
+                      // define what happens when promise is fulfilled - resolve = fulfilled (1 of 3 promise states) - responseText returns the text received by the server. resolve(fulfillmentValue)
+                       resolve(xhr.responseText);
+                    } else {
+                      // define what happens when promise is rejected - reject = rejected ( promise state) - the string will be caught in `.catch()` method
+                        reject('request rejected');
+                    }
                 }
-                // preview
-                if(data.items[i].volumeInfo.previewLink) {
-                    results.innerHTML += "<a target=\"_blank\" rel=\"noopener noreferrer\" href =" + data.items[i].volumeInfo.previewLink + "> Check out a preview <a/>"
-                } else {
-                    results.innerHTML +=  "<p>Preview Link: Sorry, this content is not available. Please try again later.</p>"
-                }
-            }
-        },
-        type:'GET'
-    })
+            });
+        // opening a request triggers a change of state and prepares browser to make request- xhr.open( httpMethod, url, async)
+        xhr.open ('GET','https://www.googleapis.com/books/v1/volumes?q=' + searchQuery, true);
+        // send request to server
+        xhr.send();
+        }
+    ));
+}
+
+const successResult = (stringifiedData) => {
+// JSON.parse() method parses (taking input from string and returning other data type)a string as JSON
+    const data = JSON.parse(stringifiedData);
```

Comment:
[Bike shedding](https://en.m.wikipedia.org/wiki/Law_of_triviality), but you could error handle here if the JSON wasn't valid.